src/plugins/intel_gpu/src/plugin/transformations/convert_gather_to_compressed.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convert_gather_to_compressed.hpp"
+#include <memory>
+
+// #include "intel_gpu/op/gather.hpp"
+#include "intel_gpu/op/gather_compressed.hpp"
+
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/pattern.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+ConvertGatherToGatherCompressed::ConvertGatherToGatherCompressed() {
+    using namespace ov::pass::pattern;
+
+    auto compressed_constant = [](const ov::Output<ov::Node>& output) {
+        return (output.get_element_type() == ov::element::u8 ||
+                output.get_element_type() == ov::element::i8 ||
+                output.get_element_type() == ov::element::u4 ||
+                output.get_element_type() == ov::element::i4) &&
+                output.get_target_inputs().size() == 1;
+    };
+
+    auto reshape_3d_to_2d = [](const ov::Output<ov::Node>& output) {
+        auto in_ps = output.get_node()->get_input_partial_shape(0);
+        auto out_ps = output.get_node()->get_output_partial_shape(0);
+        return in_ps.rank().is_static() && out_ps.rank().is_static() && in_ps.size() == 3 && out_ps.size() == 2;
+    };
```

Comment:
I am curious about the reshape. 
Is there any reason that we should assume to 3d to 2d? 
Can't it be any reshape?

Reply:
The shape of input dictionary of embedding layer is 2d and is converted to 3d by weights compression logic in NNCF. It is required to reshape it to the original state to process `Gather`.